discord/ext/commands/bot.py

```diff
@@ -141,6 +141,19 @@ def __init__(self, command_prefix, help_command=_default, description=None, **op
         if self.owner_ids and not isinstance(self.owner_ids, collections.abc.Collection):
             raise TypeError(f'owner_ids must be a collection not {self.owner_ids.__class__!r}')
 
+        self_bot = options.get('self_bot', False)
+        user_bot = options.get('user_bot', False)
+
+        if self_bot and user_bot:
+            raise TypeError('Both self_bot and user_bot are set')
+
+        if self_bot:
+            self._skip_check = lambda x, y: x != y
+        elif user_bot:
+            self._skip_check = lambda *_: False
+        else:
+            self._skip_check = lambda x, y: x == y
+
```

Comment:
Could you please explain this block? And possibly add a comment to make it easier for future development :)

Edit: Just read a bit further and read that x and y in the lambda refer to user IDs, I don't fully understand what it's supposed to do though

Reply:
If the function returns True, the command is not run.